### DISCUSSION TITLE
Thoughts on moving UniqueName to settings?

### DISCUSSION THREAD

**Discussion by schneidenbach** [tomasfabian/Joker#15] [Ideas]
Hey @tomasfabian, love your library here.

I'm using SqlTableDependencyProvider<T> to listen for changes on a specific table using the UniqueScope lifetime scope. I noted that this specific scope is designed to persist between application restarts, so I started digging into how the subscription names itself and found the following line:

https://github.com/tomasfabian/Joker/blob/08b1567c66fde3e774f53ecd5c3d7f61c971251b/SqlTableDependency.Extensions/SqlTableDependencyWithReconnection.cs#L84

In order to override this setting, I need to:
* inherit from SqlTableDependencyWithUniqueScope and override the UniqueName prop
* override CreateSqlTableDependency in SqlTableDependencyProvider<T> and pass this SqlTableDependencyWithUniqueScope in instead

It's not a ton of friction, but I was curious to know how you felt about exposing the UniqueName property in SqlTableDependencySettings<T> instead? If so, I'll issue a PR with that change.

**Comment by tomasfabian** [comment]
Hi Spencer,
thank you for reaching me out and for your appreciation.

Your suggestion sounds very reasonable to me, too. It would be great if you
will Create a PR. It would be nice to do it without a breaking change.

Do you already have a good name for this setting eg.
SqlTableDependencySettings.Scope(Unique)Name?

Regards, Tomas.


po 12. 7. 2021 o 22:13 Spencer Schneidenbach ***@***.***>
napísal(a):

**Comment by tomasfabian** [comment]
hi @schneidenbach,
I pushed the requested feature (UniqueName to settings). Could you please review it?

[Commit details](https://github.com/tomasfabian/Joker/commit/7d4f144451faf59ba78a0a1a5e5b13d044d4da75)

Thank you Tomas

**Reply by sergiu-enache** [reply]
Settings.FarServiceUniqueName does not seem to work in OnCreateSettings(), for me is still using Environment.MachineName (which in docker swarm is the container id)

**Reply by tomasfabian** [reply]
Hi @sergiu-enache, which LifetimeScope are you using?

```C#
  public enum LifetimeScope
  {
    ConnectionScope = 1,
    UniqueScope = 2,
    ApplicationScope = 3,
  }
```

**Reply by sergiu-enache** [reply]
@tomasfabian is UniqueScope, using OPs workaround I managed to make it work

**Reply by tomasfabian** [reply]
@sergiu-enache great! It should be configured in [this way](https://github.com/tomasfabian/Joker/wiki/How-to-set-additional-SqlTableDependency-settings).

What does the OPs abbreviation mean, please?

**Reply by sergiu-enache** [reply]
sorry, too much reddit haha.. Original Poster

I think I used OnCreateSettings() correctly but no luck, still used Environment.MachineName in the end.

**Reply by sergiu-enache** [reply]
also thank you for this package, CreateBulkRecordChangesNotifier will solve my issue
